File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int insertPos = 0;
        int takePos = 0;
        int k = 1;
        
        while (takePos < nums.size()) {
            if (nums[insertPos] == nums[takePos]) {
                takePos++;
            } else {                
                k++;
                nums[insertPos + 1] = nums[takePos];
                insertPos++;
                takePos++;
            }
        }
        return k;   
    }
};

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        k = 1
        
        l = 0
        r = 0
        
        while r < len(nums):
            if nums[l] == nums[r]:
                r+=1
            else:
                nums[l + 1] = nums[r]
                l += 1
                r += 1
                k += 1
        return k
        